aiidalab_widgets_base/viewers.py

```diff
@@ -318,7 +318,7 @@ def change_camera(change):
     @tl.observe("cell")
     def _observe_cell(self, _=None):
         # only update cell info when it is a 3D structure.
-        if self.cell and all(self.structure.pbc):
+        if self.cell and any(self.structure.pbc):
```

Comment:
The function `_observe_cell` is for the 3D structure with the cell, so from my understanding, even the structure that is non-periodic can be in the cell and need to go through this cell. 
In this case, an additional mapping is required from `False, False, False` to non-periodic maybe represented as `-`.

Reply:
I don't understand, why would a non-periodic structure have a cell? As a workaround for gas phase molecules in plane wave codes?

Reply:
Yep, for the plane-wave basis. I won't regard it as a workaround, it may be not as accurate as Gaussian basis but a quite standard and common calculation in "our" daily life 😄

Reply:
Haha, sure, didn't mean to offend. :D

Reply:
@unkcpz  So perhaps the method should be modified? one thing is to check if the structure has a cell (and display or not the cell parameters defined), and another one should be to inspect what periodicity has defined.